src/main/ruby/core/thread.rb

```diff
@@ -264,25 +264,21 @@ def name=(val)
   end
 
   # Java goes from 1 to 10 (default 5), Ruby from -3 to 3 (default 0)
-  PRIORITIES_RUBY_TO_JAVA = {
-    -3 => 1, -2 => 2, -1 => 4, 0 => 5,
-     1 => 7,  2 => 8,  3 => 10
-  }.freeze
-  PRIORITIES_JAVA_TO_RUBY = {
-    1 => -3, 2 => -2, 3 => -1, 4 => -1, 5 => 0,
-    6 =>  1, 7 =>  1, 8 =>  2, 9 => 2, 10 => 3
-  }.freeze
+  #                         -3 -2 -1  0  1  2   3
+  PRIORITIES_RUBY_TO_JAVA = [1, 2, 4, 5, 7, 8, 10]
```

Comment:
Your commit comment indicates why you're doing this, but I think a cautionary comment would be good here. The old code was much easier to read and I suspect someone might try to change this is the future.

Reply:
The gate should catch it with #1120 (notably `rehash` on a frozen Hash doesn't work so well).
But indeed, I can add a comment.

Reply:
Okay. I didn't realize we had a way of catching such issues.